packages/babel-cli/src/babel/dir.ts

```diff
@@ -239,7 +240,14 @@ export default async function ({
 
       try {
         const written = await Promise.all(
-          filenames.map(filename => handleFile(filename, getBase(filename))),
+          filenames.map(filename => {
+            if (outputtedFiles.has(filename)) {
+              console.error(
```

Comment:
Is this always going to end up in an infinite loop? If so, we should throw.

Reply:
I guess there is a high probability that it will fall into an endless loop.